CONTRIBUTING.md

```diff
@@ -147,33 +142,59 @@ pre-commit run --files */**
 
 Tests are run with Go's default test runner wrapped in Makefile targets. So, for example, you can run all tests with:
 
-```sh {"id":"01HF7BT3HEQBTBM9SSTS88ZSCF"}
-make test
+```sh {"id":"01HF7BT3HEQBTBM9SSTS88ZSCF","name":"test","terminalRows":"15"}
+go clean -testcache
+TAGS="test_with_docker" make test
 ```
 
 Please notice that our tests include integration tests which depend on additional software like Python or node.js. If you don't want to install them or tests fail because of different versions, you can run all tests in a Docker container:
 
-```sh
+```sh {"id":"01J5P9MKFZ4SRS1VE6JBKG3EAK","terminalRows":"15"}
 make test-docker
 ```
 
+## Development
+
+To run the server in dev mode with predictable port and TLS credentials location:
+
+```sh {"background":"true","id":"01HJ9NZT45B03C9J6YJJJDB8PG","interactive":"true","name":"server-dev"}
+go run . server --dev --address 127.0.0.1:9999 --tls /tmp/runme/tls 2>&1
+```
+
+### Upgrading Minor Version Dependencies
+
+For upgrading dependencies that have minor version changes, as well as test dependencies, use the following command. This command fetches the latest minor or patch versions of the modules required for building the current module, including their dependencies:
+Periodically dependencies need to be upgraded. For minor versions with test deps:
+
+```sh {"id":"01HF7BT3HEQBTBM9SSSG798S2D"}
+$ go get -t -u ./...
+```
+
+### Upgrading Major Version Dependencies
+
+When you need to upgrade to major versions of your dependencies, it’s prudent to upgrade each dependency one at a time to manage potential breaking changes efficiently. The `gomajor` tool can assist you in listing and upgrading major version dependencies. To list all major version upgrades available for your project, use the following command:
+
+```sh {"id":"01HF7BT3HEQBTBM9SSSK6JMS58"}
+$ gomajor list
+```
+
 ### Coverage
 
 In order to generate a coverage report, run tests using
 
-```sh {"name":"coverage-run"}
+```sh {"id":"01J5P9MKFZ4SRS1VE6JCT00N4K"}
 make test-coverage
 ```
 
 And then:
```

Comment:
What do you think about updating this paragraph?

![image](https://github.com/user-attachments/assets/aef5c6c2-0458-4e7c-8765-713bbfaf8020)